packages/react-router/modules/__tests__/useRouteMatch-test.js

```diff
@@ -1,36 +1,60 @@
+import { renderHook } from "@testing-library/react-hooks";
 import React from "react";
-import ReactDOM from "react-dom";
-import { MemoryRouter, useRouteMatch } from "react-router";
-
-import renderStrict from "./utils/renderStrict.js";
+import {
+  MemoryRouter,
+  useRouteMatch,
+  __RouterContext as RouterContext
+} from "react-router";
 
 describe("useRouteMatch", () => {
-  const node = document.createElement("div");
-
-  afterEach(() => {
-    ReactDOM.unmountComponentAtNode(node);
-  });
+  const defaultWrapper = ({ children }) => (
+    <MemoryRouter initialEntries={["/home"]}>{children}</MemoryRouter>
+  );
 
   it("returns the match object", () => {
-    let match;
-
-    function HomePage() {
-      match = useRouteMatch("/home");
-      return null;
-    }
-
-    renderStrict(
-      <MemoryRouter initialEntries={["/home"]}>
-        <HomePage />
-      </MemoryRouter>,
-      node
+    const match = { path: "/", url: "/", params: {}, isExact: true };
+    const wrapper = ({ children }) => (
+      <RouterContext.Provider
+        value={{
+          match,
+          location: { pathname: match.path }
+        }}
+      >
+        {children}
+      </RouterContext.Provider>
```

Comment:
This should continue to use MemoryRouter, not the context directly. The internal implementation could change at any time, so this couples the test to non-semver APIs, which makes them brittle.

Reply:
Makes sense, I've changed it to use MemoryRouter.